pkg/agent/util/net.go

```diff
@@ -100,47 +100,66 @@ func dialUnix(address string) (net.Conn, error) {
 	return net.Dial("unix", address)
 }
 
-// GetIPNetDeviceFromIP returns a local IP/mask and associated device from IP.
-func GetIPNetDeviceFromIP(localIP net.IP) (*net.IPNet, *net.Interface, error) {
+// GetIPNetDeviceFromIP returns local IPs/masks and associated devices from IP.
+func GetIPNetDeviceFromIP(localIPs []net.IP) (v4IPNet *net.IPNet, v6IPNet *net.IPNet, iface *net.Interface, err error) {
 	linkList, err := net.Interfaces()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	for _, link := range linkList {
-		addrList, err := link.Addrs()
+	for i := range linkList {
+		addrList, err := linkList[i].Addrs()
 		if err != nil {
 			continue
 		}
 		for _, addr := range addrList {
 			if ipNet, ok := addr.(*net.IPNet); ok {
-				if ipNet.IP.Equal(localIP) {
-					return ipNet, &link, nil
+				for _, ip := range localIPs {
+					if ipNet.IP.Equal(ip) {
+						iface = &linkList[i]
+						if ip.To4() == nil {
+							v6IPNet = ipNet
+						} else {
+							v4IPNet = ipNet
+						}
+					}
```

Comment:
is it possible for the 2 different IPs for the Node (v4 / v6) to correspond to 2 different interfaces?

the function comment talks about "associated **devices** from IP", but we only return a single interface it seems

Reply:
Discussed with Wenying, we think that maybe it is unnecessary to consider such case. In such case, user should consider have 2 clusters with such interfaces.

The comment is updated.

Reply:
I think the restriction makes sense. Especially in a scenario where we may need to move the uplink to the OVS bridge. However, do you think we can log a clean error in that case and fail agent initialization?